query-engine/connectors/sql-query-connector/src/database/operations/write.rs

```diff
@@ -453,14 +454,22 @@ pub(crate) async fn delete_records(
     let row_count = if record_filter.has_selectors() {
         let ids: Vec<_> = record_filter.selectors.as_ref().unwrap().iter().collect();
         let mut row_count = 0;
+        let mut remaining_limit = limit;
 
-        for delete in write::delete_many_from_ids_and_filter(model, ids.as_slice(), filter_condition, ctx) {
+        for delete in
+            write::delete_many_from_ids_and_filter(model, ids.as_slice(), filter_condition, remaining_limit, ctx)
+        {
             row_count += conn.execute(delete).await?;
+            if let Some(limit) = remaining_limit {
+                remaining_limit = Some(limit - row_count as i64);
+                if remaining_limit.unwrap() <= 0 {
+                    break;
+                }
```

Comment:
Let's try `unwraps` is possible, as they don't come for free.

Store `limit - row_count as i64` in both a temporary variable and the `remaining_limit` variable in the outer scope, check whether that temporary value is `<= 0`, and if so break.

Reply:
Thanks - wasn't aware of the costs of unwraps 👍